test/test_route_concat.cpp

```diff
@@ -172,7 +172,7 @@ void test_route_2_buffers(void)
     small::shape_type output_shape({1, 48, 13, 13});
     small::Tensor<Buffer> output(input0.size() + input1.size());
 
-    small::RouteLayer<Buffer> route(shape0, shape1);
+    small::RouteLayer<Buffer> route(shape0, shape1, {0});
```

Comment:
Should RouteLayer throw an exception if the number of parent ids does not match the number of input buffers?

Reply:
I added this check within the route layer ctor.